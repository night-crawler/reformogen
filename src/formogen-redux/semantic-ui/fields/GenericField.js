import { connect } from 'react-redux';
import { createStructuredSelector } from 'reselect';

import { GenericField as GenericFieldComponent } from '../../../formogen/semantic-ui/fields/GenericField';

// import { } from '../../selectors';
// import { bootstrap } from '../../actions';


function mapDispatchToProps(dispatch) {
  return {
    dispatch,
  };
}

function mergeProps(stateProps, dispatchProps, ownProps) {
  const { dispatch } = dispatchProps;

  return {
    ...ownProps,
    ...stateProps,
    ...dispatchProps,

    dispatch: undefined,
  };
}

export const GenericField = connect(
  createStructuredSelector({

  }),
  mapDispatchToProps,
  mergeProps,
)(GenericFieldComponent);
