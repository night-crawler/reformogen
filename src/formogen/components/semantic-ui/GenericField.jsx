import React from 'react';
import loglevel from 'loglevel';
import { Form } from 'semantic-ui-react';
import _ from 'lodash';

import { MessageList } from './MiscComponents';

import Label from './Label';

import propTypes from '../fieldPropTypes';


/**
 * Generic Field Component to use in DEBUG MODE ONLY
 * @param {Object} props
 * @returns {XML}
 * @constructor
 */
GenericField.propTypes = propTypes;
export default function GenericField(props) {
    loglevel
        .getLogger('Formogen/components/semantic-ui/GenericField.jsx')
        .warn(`Using a placeholder GenericFieldComponent to render field "${props.name}" of type "${props.type}"`);

    let _props = Object.assign({}, props);
    if (_.isFunction(props.updateProps)) {
        _props = props.updateProps(_props, props);
    }

    return (
        <Form.Field
            required={ props.required }
            disabled={ !props.editable }
            width={ props.layoutOpts.width }
            error={ !_.isEmpty(props.errors) }
        >
            <Label { ...props } />
            <pre>
                { JSON.stringify(_props, null, 4) }
            </pre>
            { !props.helpTextOnHover ? <span className="help-text">{ props.help_text }</span> : ''}
            <MessageList messages={ props.errors }/>
        </Form.Field>
    );
}
