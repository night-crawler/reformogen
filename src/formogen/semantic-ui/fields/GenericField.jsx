import React from 'react';
import _ from 'lodash';
import { Form } from 'semantic-ui-react';

import propTypes from '../../fieldPropTypes';
import { FieldLabel } from '../common/FieldLabel';
import { ErrorsList } from '../common/ErrorsList';


/**
 * Generic Field Component to use in DEBUG MODE ONLY
 * @param {Object} props
 * @returns {XML}
 * @constructor
 */
GenericField.propTypes = propTypes;
export function GenericField(props) {
  // eslint-disable-next-line
  console.warn(`Using a placeholder GenericFieldComponent to render field "${ props.name }" of type "${ props.type }"`);

  let _props = Object.assign({}, props);
  if (_.isFunction(props.updateProps)) {
    _props = props.updateProps(_props, props);
  }

  return (
    <Form.Field
      required={ props.required }
      disabled={ !props.editable }
      width={ props.layoutOpts.width }
      error={ !_.isEmpty(props.errors) }
    >
      <FieldLabel { ...props } />
      <pre>
        { JSON.stringify(_props, null, 4) }
      </pre>
      { !props.helpTextOnHover
        ? <span className='help-text'>{ props.help_text }</span>
        : ''
      }
      <ErrorsList messages={ props.errors } />
    </Form.Field>
  );
}
