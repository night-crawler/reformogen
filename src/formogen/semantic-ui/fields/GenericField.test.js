import GenericField from './GenericField';


it('should render GenericField', () => {
  const wrapper = shallow(
    <GenericField
      type='GenericField'
      name='GenericField: name'
      verbose_name='GenericField: verbose_name'
      help_text='GenericField: help_text'
      layoutOpts={ { width: 1 } }
    />
  );
  expect(wrapper).toMatchSnapshot();
});
