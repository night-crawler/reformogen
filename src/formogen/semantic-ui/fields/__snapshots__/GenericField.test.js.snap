// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render GenericField 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <GenericField
    help_text="GenericField: help_text"
    layoutOpts={
      Object {
        "width": 1,
      }
    }
    name="GenericField: name"
    type="GenericField"
    verbose_name="GenericField: verbose_name"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "function",
    "props": Object {
      "children": Array [
        <FieldLabel
          help_text="GenericField: help_text"
          layoutOpts={
            Object {
              "width": 1,
            }
          }
          name="GenericField: name"
          type="GenericField"
          verbose_name="GenericField: verbose_name"
        />,
        <pre>
          {
    "type": "GenericField",
    "name": "GenericField: name",
    "verbose_name": "GenericField: verbose_name",
    "help_text": "GenericField: help_text",
    "layoutOpts": {
        "width": 1
    }
}
        </pre>,
        <span
          className="help-text"
        >
          GenericField: help_text
        </span>,
        <ErrorsList
          color="red"
          header="Errors"
        />,
      ],
      "disabled": true,
      "error": false,
      "required": undefined,
      "width": 1,
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "help_text": "GenericField: help_text",
          "layoutOpts": Object {
            "width": 1,
          },
          "name": "GenericField: name",
          "type": "GenericField",
          "verbose_name": "GenericField: verbose_name",
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "{
    \\"type\\": \\"GenericField\\",
    \\"name\\": \\"GenericField: name\\",
    \\"verbose_name\\": \\"GenericField: verbose_name\\",
    \\"help_text\\": \\"GenericField: help_text\\",
    \\"layoutOpts\\": {
        \\"width\\": 1
    }
}",
        },
        "ref": null,
        "rendered": "{
    \\"type\\": \\"GenericField\\",
    \\"name\\": \\"GenericField: name\\",
    \\"verbose_name\\": \\"GenericField: verbose_name\\",
    \\"help_text\\": \\"GenericField: help_text\\",
    \\"layoutOpts\\": {
        \\"width\\": 1
    }
}",
        "type": "pre",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "GenericField: help_text",
          "className": "help-text",
        },
        "ref": null,
        "rendered": "GenericField: help_text",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "color": "red",
          "header": "Errors",
          "messages": undefined,
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
    ],
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "function",
      "props": Object {
        "children": Array [
          <FieldLabel
            help_text="GenericField: help_text"
            layoutOpts={
              Object {
                "width": 1,
              }
            }
            name="GenericField: name"
            type="GenericField"
            verbose_name="GenericField: verbose_name"
          />,
          <pre>
            {
    "type": "GenericField",
    "name": "GenericField: name",
    "verbose_name": "GenericField: verbose_name",
    "help_text": "GenericField: help_text",
    "layoutOpts": {
        "width": 1
    }
}
          </pre>,
          <span
            className="help-text"
          >
            GenericField: help_text
          </span>,
          <ErrorsList
            color="red"
            header="Errors"
          />,
        ],
        "disabled": true,
        "error": false,
        "required": undefined,
        "width": 1,
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "help_text": "GenericField: help_text",
            "layoutOpts": Object {
              "width": 1,
            },
            "name": "GenericField: name",
            "type": "GenericField",
            "verbose_name": "GenericField: verbose_name",
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "{
    \\"type\\": \\"GenericField\\",
    \\"name\\": \\"GenericField: name\\",
    \\"verbose_name\\": \\"GenericField: verbose_name\\",
    \\"help_text\\": \\"GenericField: help_text\\",
    \\"layoutOpts\\": {
        \\"width\\": 1
    }
}",
          },
          "ref": null,
          "rendered": "{
    \\"type\\": \\"GenericField\\",
    \\"name\\": \\"GenericField: name\\",
    \\"verbose_name\\": \\"GenericField: verbose_name\\",
    \\"help_text\\": \\"GenericField: help_text\\",
    \\"layoutOpts\\": {
        \\"width\\": 1
    }
}",
          "type": "pre",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "GenericField: help_text",
            "className": "help-text",
          },
          "ref": null,
          "rendered": "GenericField: help_text",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "color": "red",
            "header": "Errors",
            "messages": undefined,
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
      ],
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
